src/components/Tabs/components/TabMeasurer/TabMeasurer.tsx

```diff
@@ -83,25 +84,31 @@ export default class TabMeasurer extends React.PureComponent<Props, never> {
   };
 
   private handleMeasurement = () => {
-    if (this.containerNode == null) {
-      return;
+    if (this.animationFrame) {
+      cancelAnimationFrame(this.animationFrame);
     }
 
-    const {handleMeasurement} = this.props;
-    const containerWidth = this.containerNode.offsetWidth;
-    const tabMeasurerNode = findDOMNode(this);
-    const hiddenTabNodes =
-      tabMeasurerNode instanceof Element && tabMeasurerNode.children;
-    const hiddenTabNodesArray: HTMLElement[] = [].slice.call(hiddenTabNodes);
-    const hiddenTabWidths = hiddenTabNodesArray.map((node) => {
-      return node.getBoundingClientRect().width;
-    });
-    const disclosureWidth = hiddenTabWidths.pop() as number;
-
-    handleMeasurement({
-      containerWidth,
-      disclosureWidth,
-      hiddenTabWidths,
+    this.animationFrame = requestAnimationFrame(() => {
+      if (this.containerNode == null) {
```

Comment:
I think the check for `containerNode` should be outside of the frame.

Reply:
I tried that as well, but you'll see a type error. I think it makes sense to check the container node just before we use it. Since we are now doing the measurements async, it's possible that the container node could change in between when we schedule and when we measure.